src/components/Registration.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import '../components/Registration.css';
+
+const Registration = () => {
+
+    const [user, setUser] = useState ({
+        username: "",
+        email: "",  
+        password: "",
+        passwordConfirm: "",
+     })
+
+     const handleEmail = (typedEmail) => {
+        if (typedEmail.includes("@")) {
+        setUser({...user, email: typedEmail});
+        }
+     }
+
+     const handleUserName = () => {
+        if (user.email !== null) {
+        const email = user.email;
+        setUser({...user, username: email.slice(0, email.indexOf("@"))})}
+        else alert("Use valid email address, please.")
+     }
```

Comment:
jak přesně má fungovat tohle? Pokud čtu co dělá ten kód tak:
- na každou změnu políčka userName (když uživatel něco napíše do políčka username)
- zkontroluj jestli email není null
- pokud není - nastav uživatelské jméno na první část emailu (před @)
- pokud je, obraz hláčku uživateli

Zkoušela sis že to funguje? (tj jméno se ti vyplní). Vidím tam pár problémů:
- vyplnění uživatelského jména na základě emailu se nemůže dít ve chvíli kdy uživatel napsal něco do uživatelského jména, protože mu to přeplácneš tím co je v emailu
- user.email nikdy nebude null (zkus si to přes console.log), i v defaultním stavu nastavuješ prázdný řetězec
- setUser ti nezmění hodnoty formuláře protože není kontrolovaný, viz komentář výše

Reply:
To máš pravdu, to byla blbost. Do username jsem dala onChange, podobný jako u ostatních polí. Předělala jsem to do funkce handleSubmit, která se spustí při odeslání formuláře, měla by kontrolovat validní email (jinak se ukončí), pokud je username prázdné, vyplnit pole částí emailu před @ (jinak tam nechá, co je vyplněné), a nakonec zjistit, že hesla se shodují (jinak se zase ukončí).